src/Playwright/Core/Frame.cs

```diff
@@ -535,6 +538,9 @@ private Waiter SetupNavigationWaiter(string apiName, float? timeout)
 
         private bool UrlMatches(string url, string matchUrl, Regex regex, Func<string, bool> match)
         {
+            var baseUrl = (Page.Context as BrowserContext)?.BaseUrl;
```

Comment:
we should only combine if its not starting with * and a string. The second condition is for match always true but I miss a * check there.

Reply:
Is there a test case that I missed for this? Also, do you mean `baseUrl` or the `url` start with a `*`? If the latter, then we *do* check:

```
 if (string.IsNullOrEmpty(baseUrl) || !Uri.IsWellFormedUriString(url, UriKind.RelativeOrAbsolute))
```

Because that won't be a well-formed URI string and we'll just return the unmerged URL.

Reply:
yes this is a test-case which was not there, let me add one.

Reply:
Added that test. And separate explicit check.